lib/utils/resolveObject.js

```diff
@@ -62,36 +66,29 @@ function traverseObj(obj) {
         traverseObj( obj[key] );
       } else {
         if (typeof obj[key] === 'string' && obj[key].indexOf('{') > -1) {
-          try {
-            obj[key] = compile_value( obj[key], [key] );
-          }
-          catch (e) {
-            errors.push(e);
-          }
+          obj[key] = compile_value( obj[key], [current_context.join('.')] );
```

Comment:
Why is the second argument changing from `[key]` to `[current_context.join('.')]`

Reply:
This is part of the bugfix. [key] was something like ['base'] instead of ['font.size.base']. In the compile_value code, this array is the traced stack of replaced values. if something later references 'font.size.base', it would not be recognized as a circular reference and fail. This behavior is exhibited in #118.

Reply:
My earlier finding that this was generating errors is no longer replicable because it was a interaction with the old compile_value code.

In its current form testing and code are both correct as written in the PR. I'll try to explain. Lets look at __tests__/json_files/circular_2.json

With [key], the compile_value function is called in the following order with the following arguments:
```
  {j} [ 'c' ]
  {a.b.c} [ 'c', 'j' ]
  {a.b.c} [ 'j' ]
```

With [current_context.join('.')], the compile_value function is called in the following order with the following arguments:
```
  {j} [ 'a.b.c' ]
  {a.b.c} [ 'j' ]
```

What does this mean?

In the first situation, 'c' is found to reference 'j' which is found to resolve to 'a.b.c', which is then found to resolve to 'j', which is a circular reference. It does catch the circular reference, but it was wrong in that it found c -> j -> a.b.c -> j. 'c' isn't actually a property.

In the second situation, 'a.b.c' is found to reference 'j' which is found to resolve to 'a.b.c', which is a circular reference. This is what we should expect.

IMO, the tests failing with [key] is actually a good thing because it ensures that we capture circular references in correct order. I don't see a way to directly write tests to find and catch that the first value in the stack is being incorrectly written. Let me know if you have any suggestions.

Reply:
I am pushing on more change to the tests to put each circular reference test in its own it('should ...') wrapper. Turns out Jest stops executing any block inside an it() after it fails an expect, which is maybe not what we want.